Inset map viz

elmeraa opened the pull request:
## Summary of changes
-  Added new targets in `2_process.R` to transform `p2_prop_western_2_StaID_metadata` by filtering to the unique staID's from `p2_prop_western_2_StaID` then converted lat/long to geometry points & included code to pull in state borders from `spData::us_states`  with additional target to filter for the 11 western states.
-  added new plotting function `plot_inset.R` that plots all state boundaries, the highlighted 11 western states, and the station and exports png with transparent background to `3_visualize/out`

### With stations 
![western_states_station_inset](https://user-images.githubusercontent.com/69014273/202330275-298f5e82-8d78-4fa4-9814-675a6de7c12a.png)
### Without
![western_states_inset](https://user-images.githubusercontent.com/69014273/202330309-cd108004-d1f4-4bcd-82b2-eb85b8d00e68.png)

Couple notes:
- line thickness could probably be bigger, especially depending on the dimensions of the inset (wasn't so sure on that)
- grey color can obviously be adjusted, and so can the alpha to make the state lines more visible
- the `ms_simplify()` on the state boundaries looks pretty stiff but probably won't matter as much when its a smaller inset

elmeraa commented:
Couple notes:
- line thickness could probably be bigger, especially depending on the dimensions of the inset (wasn't so sure on that)
-  grey color can obviously be adjusted, and so can the alpha to make the state lines more visible

hcorson-dosch reviewed (changes requested):
👍👍 _Really_ nice work on this Elmera, and thank you for getting this PR up so quickly. Your targets set up is neat and clean, and it ran perfectly for me locally.

After testing it in the site I've requested a few changes (comments prefaced with 'RC'). I've also made a couple suggested changes (prefaced with 'SC') that you can take or leave, and left a couple of general comments ('GC') with food for thought on pipeline structure.

elmeraa commented:
## Summary of requested changes
-  moved targets from `2_process.R` to `3_visualize` - I agree, it does make more sense there
- reduced the need for excess targets that are not used downstream such as `p2_prop_western_2_StaID_metadata_sf` and  updated `p2_states_western` to reduce the need of extra code.
- added if statement to `plot_inset` function that displays stations on map when `stations =TRUE` and added additional target `p3_inset_stations_map` for seperatly outputting inset with stations displayed.
- additional aesthetic, coloring, and sizing edits

## With stations
![western_states_stations_inset](https://user-images.githubusercontent.com/69014273/202554620-65246274-b5f3-4ff9-a6c8-bdda3b6c39ae.png)

## Without Stations 
![western_states_inset](https://user-images.githubusercontent.com/69014273/202554658-b1b2513d-215d-4142-80db-63c07ee781b3.png)

elmeraa commented on 3_visualize/src/plot_inset.R:
```diff
@@ -0,0 +1,29 @@
+            fill = NA,
+            color = 'black') +
+    geom_sf(data = western_us_data,
+            fill = '#DDE2E3',
```
without `alpha = 0.5`, it ends up being a bit too dark. 
<img width="1440" alt="Screen Shot 2022-11-17 at 12 30 12 PM" src="https://user-images.githubusercontent.com/69014273/202554928-98acd6a4-e81d-4a75-8b23-826ef427a083.png">

hcorson-dosch reviewed (approved):
🎉 Looks great! Thanks for making those edits. I think you got the simplification on the states just right 👌

Down the road we may decide to switch all or part of the map to svg, but TBD on that.
